2-Intelligent-Agents/c_cleaningRobot.js

```diff
@@ -233,7 +234,144 @@ function makeTableControlledDiagram() {
     }
 }
 
+/*Control the daigram based on the performance parameters set 
+by the reader that the AI is supposed to follow.The animation flow 
+is similar to the first agent controlled diagram but there is an 
+additional table UI that lets the reader view the percepts and actions 
+being followed as well as change the rules followed by the agent.
+*/
+function makePerformanceControlledDaigram(){
+    //variable declarations for the agents
+    let diagram_agent1 = makeDiagram('#performance-controlled-diagram-agent1 svg');
+    let diagram_agent2 = makeDiagram('#performance-controlled-diagram-agent2 svg');
+    var parameters = getRulesFromPage(); //reader defined agent parameters
+    var score_agent1 = 0;
+    var score_agent2 = 0;
+    var performance_agent1 = [];
+    var performance_agent2 = [];
+    var agent2_interval;    
+
+    //update agent1's environment
+    function update_agent1() {
```

Comment:
The code for updating agent1 and the code for updating agent2 are almost the same. These two should use shared code.